plugins/builds/create.js

```diff
@@ -46,10 +46,14 @@ module.exports = () => ({
                         // check if user has push access
                         .then((permissions) => {
                             if (!permissions.push) {
+                                // the user who are not permitted is deleted from admins table
+                                delete pipeline.admins[username];
                                 throw boom.unauthorized(`User ${username} `
                             + 'does not have push permission for this repo');
                             }
                         })
+                        // user has good permissions, add the user as an admin
+                        .then(() => { pipeline.admins[username] = true; })
```

Comment:
Do you need to call `pipeline.update()` after this line and before the `pipeline.sync()`? Like here: https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/pipelines/update.js#L92-L93

Reply:
I need it. I add `update` to call `sync`.  Thank you!